Examples/Scripts/Python/ckf_tracks.py

```diff
@@ -1,12 +1,126 @@
+
+    Parameters
+    ----------
+    s: Sequencer
+        the sequencer module to which we add the Seeding steps (returned from addSeeding)
+    trackingGeometry : tracking geometry
+    field : magnetic field
+    outputDirCsv : Path|str, path, None
+        the output folder for the Csv output, None triggers no output
+    outputDirRoot : Path|str, path, None
+        the output folder for the Root output, None triggers no output
+    """
+
+    if int(s.config.logLevel) <= int(acts.logging.DEBUG):
+        acts.examples.dump_args_calls(locals())
+
+    # Setup the track finding algorithm with CKF
+    # It takes all the source links created from truth hit smearing, seeds from
+    # truth particle smearing and source link selection config
+    trackFinder = acts.examples.TrackFindingAlgorithm(
+        level=s.config.logLevel,
+        measurementSelectorCfg=acts.MeasurementSelector.Config(
+            [(acts.GeometryIdentifier(), ([], [15.0], [10]))]
+        ),
+        inputMeasurements="measurements",
+        inputSourceLinks="sourcelinks",
+        inputInitialTrackParameters="estimatedparameters",
+        outputTrajectories="trajectories",
+        findTracks=acts.examples.TrackFindingAlgorithm.makeTrackFinderFunction(
+            trackingGeometry, field
+        ),
+    )
+    s.addAlgorithm(trackFinder)
+
+    if outputDirRoot is not None:
+        outputDirRoot = Path(outputDirRoot)
+        if not outputDirRoot.exists():
+            outputDirRoot.mkdir()
+
+        # write track states from CKF
+        trackStatesWriter = acts.examples.RootTrajectoryStatesWriter(
+            level=s.config.logLevel,
+            inputTrajectories=trackFinder.config.outputTrajectories,
+            # @note The full particles collection is used here to avoid lots of warnings
+            # since the unselected CKF track might have a majority particle not in the
+            # filtered particle collection. This could be avoided when a seperate track
+            # selection algorithm is used.
+            inputParticles="particles_selected",
+            inputSimHits="simhits",
+            inputMeasurementParticlesMap="measurement_particles_map",
+            inputMeasurementSimHitsMap="measurement_simhits_map",
+            filePath=str(outputDirRoot / "trackstates_ckf.root"),
+            treeName="trackstates",
+        )
+        s.addWriter(trackStatesWriter)
+
+        # write track summary from CKF
+        trackSummaryWriter = acts.examples.RootTrajectorySummaryWriter(
+            level=s.config.logLevel,
+            inputTrajectories=trackFinder.config.outputTrajectories,
+            # @note The full particles collection is used here to avoid lots of warnings
+            # since the unselected CKF track might have a majority particle not in the
+            # filtered particle collection. This could be avoided when a seperate track
+            # selection algorithm is used.
+            inputParticles="particles_selected",
```

Comment:
The comment says it's the 'full particles collection' but it's hard coded to `particles_selected`. That comes from FATRAS' `preselectParticles`?

Reply:
Right, `particles_selected` comes from the output of Fatras' initial `ParticleSelector`. At the moment, the selector passes all particles, so it shouldn't make any difference. We run the selector anyway to maintain exact consistency for the python checks, but we could drop that later just for speed.

I don't know if this comment is referring to the "full particles collection" as the particles before the Fatras preselection (`ParticleSelector`), or the particles before the seeding preselection (`TruthSeedSelector`). If the latter, then the comment still makes sense.

Maybe worth pointing out, that this setting and the comment are unchanged by this PR. It was just moved to `addCKFTracks` and hardcoded here, rather than in `selector`.

Reply:
Hm. Ok I don't remember where this is from then. I don't think I wrote it 😅